src/daily_output.ts

```diff
@@ -0,0 +1,432 @@
+import { readFileSync, writeFileSync } from 'fs';
+import { GROUP_EFFORT_REWARDS, ONE_THOUSAND_MB, OUTPUT_NAME, OUTPUT_TEMPLATE_NAME } from './constants';
+import { GQLEdgeInterface } from './gql_types';
+import { OutputData, StakedPSTHolders, WalletsStats, WalletStatEntry } from './inferno_types';
+
+/**
+ * A class responsible of parsing the GQL and CommunityOracle data into the OutputData file
+ */
+export class DailyOutput {
+	private data = this.read();
+	private latestBlock = 0;
+	private latestTimestamp = 0;
+	private bundlesTips: { [address: string]: number } = {};
+
+	/**
+	 * Takes the data from the previously generated data, fallbacking to the base template if not present
+	 * @throws if the validation of the output file fails
+	 * @retuns {OutputData}
+	 */
+	public read(): OutputData {
+		const data = (() => {
+			try {
+				return this.readOutputFile();
+			} catch (err) {
+				console.info(`The output file hasn't yet been created. Using the file template`);
+				return this.readTemplate();
+			}
+		})();
+		if (!this.validateDataStructure(data)) {
+			throw new Error(`The output JSON has a wrong structure`);
+		}
+		return data;
+	}
+
+	/**
+	 * @param {StakedPSTHolders} stakedPSTHolders key/value of address/tokens above 200 ARDRIVE locked for at least 21600 blocks (~30 days)
```

Comment:
IDE nit: If you move this helpful comment to a doc block on the `StakedPSTHolders` interface, you will see the description when you hover over the variable in the UI

Reply:
da7cdcbfcdcb7686fd3fed0f3d503246ab02f06c